dev/com.ibm.ws.sipcontainer/src/com/ibm/ws/sip/stack/transport/sip/netty/SipUdpConnLink.java

```diff
@@ -0,0 +1,492 @@
+		 * AdminHelper.getPlatformHelper().isZOS();
+		 */
+		m_sendThread = new SendThread(this);
+		m_connected = false;
+		m_outboundBuffer = null;
+		m_sendThread.start();
+
+		// read configuration settings for outbound UDP
+		SipPropertiesMap config = ApplicationProperties.getProperties();
+		m_receiveBufferSizeSocket = config.getString(StackProperties.RECEIVE_BUFFER_SIZE_SOCKET);
+		if (m_receiveBufferSizeSocket.equals("") || m_receiveBufferSizeSocket.length() == 0) {
+			m_receiveBufferSizeSocket = null;
+		}
+		m_sendBufferSizeSocket = config.getString(StackProperties.SEND_BUFFER_SIZE_SOCKET);
+		if (!m_sendBufferSizeSocket.equals("") || m_sendBufferSizeSocket.length() == 0) {
+			m_sendBufferSizeSocket = null;
+		}
+		m_receiveBufferSizeChannel = config.getString(StackProperties.RECEIVE_BUFFER_SIZE_CHANNEL);
+		if (m_receiveBufferSizeChannel.equals("") || m_receiveBufferSizeChannel.length() == 0) {
+			m_receiveBufferSizeChannel = null;
+		}
+		if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+			Tr.debug(this, tc, "<init>",
+					"receiveBufferSizeSocket [" + m_receiveBufferSizeSocket + "] sendBufferSizeSocket ["
+							+ m_sendBufferSizeSocket + "] receiveBufferSizeChannel [" + m_receiveBufferSizeChannel
+							+ ']');
+		}
+	}
+
+	/**
+	 * called by the channel while the channel factory instantiates the new
+	 * OutboundVirtualConnection object
+	 * 
+	 * @return the conn link instance associated with this connection
+	 */
+	static SipUdpConnLink getPendingConnection() {
+		SipUdpConnLink current = s_connectingInstance;
+		s_connectingInstance = null;
+		return current;
+	}
+
+	/**
+	 * establishes the outbound virtual connection
+	 */
+	private void connect(MessageContext messageContext) throws IOException {
+		String outboundChainName = m_channel.getOutboundChainName();
+		if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+			Tr.debug(this, tc, "<connect>", "outboundChainName = " + outboundChainName);
+		}
+		// TODO ANNA		
+		String localHostname = null; 
+		int localPort = 0;
+		// use the selected interface for the outbound connection
+		if (messageContext != null && messageContext.getSipConnection() != null) {
+			localHostname = messageContext.getSipConnection().getSIPListenningConnection().getListeningPoint()
+					.getHost(); // can't use the real port because we're already listening on it
+		}
+//		UDPRequestContext connectRequestContext =
+//			UDPRequestContextFactory.getRef().createUDPRequestContext( 
+//					localHostname, localPort); 
+//        if (_localHostName != null && !_localHostName.equals("*")) {
+//            this.localAddress = new InetSocketAddress(_localHostName, _localPort);
+//        } else {
+//            this.localAddress = new InetSocketAddress(_localPort);
+//        }
+
+		
+		if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+			Tr.debug(this,  tc,"<connect>","connectAsynch...");
+		}
+		
+		// TODO connect async
```

Comment:
TODO

Reply:
Currently no async code here, but should there be eventually?